features/netsocket/SocketAddress.cpp

```diff
@@ -215,17 +215,17 @@ void SocketAddress::set_port(uint16_t port)
 
 const char *SocketAddress::get_ip_address() const
 {
-    char *ip_address = (char *)_ip_address;
-
-    if (!ip_address[0]) {
+    if (!_ip_address[0]) {
         if (_addr.version == NSAPI_IPv4) {
-            ipv4_to_address(ip_address, _addr.bytes);
+            ipv4_to_address(_ip_address, _addr.bytes);
         } else if (_addr.version == NSAPI_IPv6) {
-            ipv6_to_address(ip_address, _addr.bytes);
+            ipv6_to_address(_ip_address, _addr.bytes);
+        } else {
```

Comment:
I was wondering what to return here. I don't think "*" is something users would expect.

"0.0.0.0" does have the benifit that it's compatible with current code and also parsable as an address.

An alternative is to simply return a null pointer, since the representation doesn't really exist.